src/launcher.fnl

```diff
@@ -90,68 +90,85 @@ If ~/.fennelrc exists, it will be loaded before launching a repl.")
+                       (table.remove arg i))
+        :--check-unused-locals (do
+                                 (set options.checkUnusedLocals true)
+                                 (table.remove arg i))
+        :--globals (do
+                     (allow-globals (table.remove arg (+ i 1)) _G)
                      (table.remove arg i))
-    :--no-compiler-sandbox (do
-                             (set options.compiler-env _G)
-                             (table.remove arg i))
-    :--raw-errors (do
-                    (set options.unfriendly true)
+        :--globals-only (do
+                          (allow-globals (table.remove arg (+ i 1)) {})
+                          (table.remove arg i))
+        :--require-as-include (do
+                                (set options.requireAsInclude true)
+                                (table.remove arg i))
+        :--skip-include (let [skip-names (table.remove arg (+ i 1))
+                              skip (icollect [m (skip-names:gmatch "([^,]+)")] m)]
+                          (set options.skipInclude skip)
+                          (table.remove arg i))
+        :--use-bit-lib (do
+                         (set options.useBitLib true)
+                         (table.remove arg i))
+        :--metadata (do
+                      (set options.useMetadata true)
+                      (table.remove arg i))
+        :--no-metadata (do
+                         (set options.useMetadata false)
+                         (table.remove arg i))
+        :--no-compiler-sandbox (do
+                                 (set options.compiler-env _G)
+                                 (table.remove arg i))
+        :--raw-errors (do
+                        (set options.unfriendly true)
+                        (table.remove arg i))
+        :--plugin (let [opts {:env :_COMPILER :useMetadata true :compiler-env _G}
+                        plugin (fennel.dofile (table.remove arg (+ i 1)) opts)]
+                    (table.insert options.plugins 1 plugin)
                     (table.remove arg i))
-    :--plugin (let [opts {:env :_COMPILER :useMetadata true :compiler-env _G}
-                    plugin (fennel.dofile (table.remove arg (+ i 1)) opts)]
-                (table.insert options.plugins 1 plugin)
-                (table.remove arg i))))
+        :--repl (set options.i (+ options.i 2))
+        :--compile (set options.i (+ options.i 2))
+        :-c (set options.i (+ options.i 2))
+        :--compile-binary (set options.i (+ options.i 2))
+        :--eval (set options.i (+ options.i 2))
+        :-e (set options.i (+ options.i 2))
+        :-v (set options.i (+ options.i 2))
+        :--version (set options.i (+ options.i 2))
+        :--help (set options.i (+ options.i 2))
+        :-h (set options.i (+ options.i 2))
+        "-" (set options.i (+ options.i 2))
+        _ (do
+            (set options.i (+ options.i 2))
+            (set options.ignore-options true))))))
```

Comment:
Thanks for submitting this; sorry for the delay. I broke my hand a few weeks ago.

Instead of every recognized command being a match clause, let's add them all to a `commands` table and then only set ignore-options to true if it's not found in the commands table. It should tidy this up a lot.

Plus instead of adding a new `if` for options.ignore-options, why not merge that into the `while` condition above?

Reply:
> Thanks for submitting this; sorry for the delay. I broke my hand a few weeks ago.


The delay is no problem since i only returned from travelling today. I hope you get well soon.

Thanks for the feedback, i made the changes you suggested in d91bfe3140354a726c459b640d8c6d46aeec9c01. Additionally i used the scope provided by the ``let`` for the ``commands`` table to replace ``options.i`` with a local ``var``.  Please let me know if there is anything else.